docs/proposals/20-ha-control-planes.md

```diff
@@ -0,0 +1,2624 @@
+                  namespaces:
+                    description: namespaces specifies a static list
+                      of namespace names that the term applies to. The
+                      term is applied to the union of the namespaces
+                      listed in this field and the ones selected by
+                      namespaceSelector. null or empty namespaces list
+                      and null namespaceSelector means "this pod's namespace"
+                    items:
+                      type: string
+                    type: array
+                  topologyKey:
+                    description: This pod should be co-located (affinity)
+                      or not co-located (anti-affinity) with the pods
+                      matching the labelSelector in the specified namespaces,
+                      where co-located is defined as running on a node
+                      whose value of the label with key topologyKey
+                      matches that of any node on which any of the selected
+                      pods is running. Empty topologyKey is not allowed.
+                    type: string
+                required:
+                - topologyKey
+                type: object
+              type: array
+          type: object
+      type: object
+  type: object
+```
+</details>
+
+#### Network Policy Changes
+
+Multi-node etcd brings in an additional communication line amongst the etcd members which should now be allowed. 
+An additional network policy needs to be created:
+
+<details>
+<summary> Etcd Peer Network Policy </summary>
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+name: allow-etcd-peer
+namespace: <shoot-ns>
+labels:
+  gardener.cloud/role: controlplane
+spec:
+  egress:
+  - ports:
+    - port: 2379
+      protocol: TCP
+    - port: 8080
+      protocol: TCP
+    - port: 2380
+      protocol: TCP
+    to:
+    - podSelector:
+        matchLabels:
+          app: etcd-statefulset
+          gardener.cloud/role: controlplane
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          app: etcd-statefulset
+          gardener.cloud/role: controlplane
+    ports:
+    - port: 2379
+      protocol: TCP
+    - port: 8080
+      protocol: TCP
+    - port: 2380
+      protocol: TCP
+  podSelector:
+    matchLabels:
+      app: etcd-statefulset
+      gardener.cloud/role: controlplane
+  policyTypes:
+  - Ingress
+  - Egress
+```
+</details>
+
+#### Configuration changes
+
+##### Timeout configuration
+
+It has been observed that network latencies across zones differes from one provider to another. Volume types that are available across providers can also differ w.r.t disk latency. It is therefore prudent to [fine tune](https://etcd.io/docs/v3.3/tuning/) `heart beat` interval and `election timeout` intervals for an etcd cluster. Performance tests against various providers needs to be conducted to determine optimal configurations for every cloud provider.
```

Comment:
That will probably be a problem in terms of “cloud provider agnosticity”. Can we find reasonable values, upper limits, etc. to avoid that?

Reply:
“Cloud provider agnosticity” wouldn't be a huge problem since the extension lib or Gardener already provides means to mutate the `etcd` resource when it's created/updated. I added the ref to the Ensurer interface to document that option we might consider if no reasonable and common limit can be found.

Reply:
Yes, all good and well, so maybe "agnosticity" isn't technically a problem, but is there really the plan to find values dependent on the cloud provider and then implement some tests or whatever to retest them continuously? Anyway, resolving the discussion.

Reply:
> but is there really the plan to find values dependent on the cloud provider and then implement some tests or whatever to retest them continuously? Anyway, resolving the discussion.

I don't think this is going to happen either. By the time of writing this GEP I think the primary intention was to document this potential requirement but we'll hopefully follow a more practical approach.